src/include/migraphx/op/roialign.hpp

```diff
@@ -42,7 +42,8 @@ struct roialign
 
     shape compute_shape(std::vector<shape> inputs) const
     {
-        check_shapes{inputs, *this}.has(3).standard();
+        check_shapes{inputs, *this}.has(3);
+        check_shapes{{inputs.at(1), inputs.at(2)}, *this}.standard();
```

Comment:
It should be inputs[0] and inputs[2] to be standard.

Reply:
Line 201 to 203, can be changed to
                std::array<float, 2> roi_starts = {
                    static_cast<float>(roi(n, 1) * spatial_scale),
                    static_cast<float>(roi(n, 0) * spatial_scale)};

Reply:
> 
From what i understand,
input[0] is input tensor.  input[1] is ROI co-ordinates and input[2] is batch indices.  So, why input[0] need to be standard ?

Reply:
If you look at the line 

```
               const auto bottom_data   = x.begin();
```
It means args[0] is read according to the offset, so the input shape should be standard.
But for args[1], it is access is read from the tensor_view [] operator, which can handle non-standard shape.